deepchem/models/torch_models/infograph.py

```diff
@@ -0,0 +1,435 @@
+            weights = weights[0]
+
+        return inputs, labels, weights
+
+    def unsup_sup_loss(self, inputs, labels, weights):
+        y, M = self.components['encoder'](inputs)
+        y_, M_ = self.components['unsup_encoder'](inputs)
+
+        g_enc = self.components['ff1'](y)
+        g_enc1 = self.components['ff2'](y_)
+
+        measure = 'JSD'
+        loss = self.global_global_loss_(g_enc, g_enc1, measure)
+        return loss
+
+    def local_global_loss_(self, l_enc, g_enc, batch):
+        """
+        Parameters:
+        ----------
+        l_enc: torch.Tensor
+            Local feature map from the encoder.
+        g_enc: torch.Tensor
+            Global features from the encoder.
+        batch: torch.Tensor
+            Batch tensor
+
+        Returns:
+        -------
+        loss: torch.Tensor
+            Local Global Loss value
+        """
+        num_graphs = g_enc.shape[0]
+        num_nodes = l_enc.shape[0]
+
+        pos_mask = torch.zeros((num_nodes, num_graphs)).to(self.device)
+        neg_mask = torch.ones((num_nodes, num_graphs)).to(self.device)
+        for nodeidx, graphidx in enumerate(batch):
+            pos_mask[nodeidx][graphidx] = 1.
+            neg_mask[nodeidx][graphidx] = 0.
+
+        res = torch.mm(l_enc, g_enc.t())
+
+        E_pos = self.get_positive_expectation(res * pos_mask, average=False)
+        E_pos = (E_pos * pos_mask).sum() / pos_mask.sum()
+        E_neg = self.get_negative_expectation(res * neg_mask, average=False)
+        E_neg = (E_neg * neg_mask).sum() / neg_mask.sum()
+
+        return E_neg - E_pos
+
+    def global_global_loss_(self, g_enc, g_enc1):
+        """
+        Parameters:
+        ----------
+        g_enc: torch.Tensor
+            Global features from the encoder.
+        g_enc1: torch.Tensor
+            Global features from the separate encoder.
+
+        Returns:
+        -------
+        loss: torch.Tensor
+            Global Global Loss value
+        """
+        num_graphs = g_enc.shape[0]
+
+        pos_mask = torch.eye(num_graphs).to(self.device)
+        neg_mask = 1 - pos_mask
+
+        res = torch.mm(g_enc, g_enc1.t())
+
+        E_pos = self.get_positive_expectation(res * pos_mask, average=False)
+        E_pos = (E_pos * pos_mask).sum() / pos_mask.sum()
+        E_neg = self.get_negative_expectation(res * neg_mask, average=False)
+        E_neg = (E_neg * neg_mask).sum() / neg_mask.sum()
+
+        return E_neg - E_pos
+
+    def get_positive_expectation(self, p_samples, average=True):
+        """Computes the positive part of a divergence / difference.
+
+        Parameters:
+        ----------
+        p_samples: torch.Tensor
+            Positive samples.
+        average: bool
+            Average the result over samples.
+
+        Returns:
+        -------
+        Ep: torch.Tensor
+            Positive part of the divergence / difference.
+        """
+        log_2 = math.log(2.)
+
+        if self.measure == 'GAN':
```

Comment:
All these different values for measure should be documented in the doctoring for the class